### ISSUE TITLE
Change Item on Drag

### ISSUE THREAD

**Issue by mfriesen** [lukasoppermann/html5sortable#296]
Similar to issue #265. I would like to change the item being dragged. IE: When dragging a &lt;li&gt; item, change it to a <input />. 

I was thinking of a adding a copyFrom option where the copyFrom is the ID of an hidden element? I only need it when the copy=true. I adding this option when the copy=false would be tricky as you would have to swap the item back, if the drop failed.

**Comment by lukasoppermann** [comment]
Hey, so you are saying, you have for e.g. a some draggable `<li>s` and on drag want to change it to an `<img>`? 

This sounds like a very specific case/edge case to me. Maybe it would be better to somehow do this in the `onDragStart` event? You could check if the `copy` option is set to true and than do it, correct? Do you think that would be a good solution as well?

If so, maybe you can add a `recipes` sections or something to the readme and add this example? 

I just want to make sure we don't add to many things and complicate the whole package. 😉 
However feel free to provide some arguments why it would be better to add this as a specific option and we can discuss it.

**Comment by lukasoppermann** [comment]
@mfriesen do you think your issue is the same as #289?

**Comment by mfriesen** [comment]
Yes, I believe so. Basically I'm creating a toolbar and when you drag the toolbar item, I need it to expand as the type of item it is dropping. I will try and see if onDragStart will work.

**Comment by mfriesen** [comment]
onDragStart is the right time but I ran into a problem. The sortstart only fires if the source sortable is connected to the drop sortable. IE: If list1 is acceptFrom: false and list2 is acceptFrom: .list1. The event will not fire. I am not entirely sure why?

For example in the acceptFrom example. If you add
```
sortable('.js-sortable-oneway')[0].addEventListener('sortstart', function(e) {
    	console.log("sortstart....")
	});
```
This will never fire.

**Comment by lukasoppermann** [comment]
Hmm, this seems to be because of `_dispatchEventOnConnected` https://github.com/lukasoppermann/html5sortable/blob/master/src/html.sortable.js#L433-L438 although I don't see the sense in it. I guess we should carefully remove it.

I would guess this function is used so that a non-connected sortable does not get the event if I drop something onto it (which will not work, because it is not connected). However, this does not really make sense for `sortstart` I think.

**Comment by mfriesen** [comment]
I agree. I think creating a _dispatchEvent function and calling that from sortstart would be fine and I don't think it would break anything. Let me know if you want me to help out.

**Comment by lukasoppermann** [comment]
Hey, @mfriesen if you could help out that would be really neat.

**Comment by mfriesen** [comment]
@lukasoppermann Do you think it makes sense to add a placeholderDrag option? Much like the current placeholder option. This would change what you are dragging? I tried to change the icon from the onDragStart but I couldn't get it to work.

**Comment by lukasoppermann** [comment]
Hey @mfriesen are you talking about the ghost? As in the half transparent thing that is shown when you drag something? E.g. #271

This could definitely make sense, however I believe it is a little bit more complicated, as you can not change it directly. What you would need to do is create an item and assign this to the ghost. Basically like this: https://github.com/lukasoppermann/nestable/blob/master/src/nestable.js#L150-L173

However, the linked code is very old and just an inspiration. 😉 

If you want to look into it, that would be very neat, but please in a separate PR (I prefer small PRs, as they are much easier & faster to review).

Thank you!

**Comment by mfriesen** [comment]
@lukasoppermann yes. that's what I want. When I have time I'll look into it.

**Comment by mfriesen** [comment]
@lukasoppermann I'm thinking just overriding the setDragImage to a custom image is enough. Something adding options for 
dragImage, dragImageWidth, dragImageHeight.

Thoughts?

**Comment by lukasoppermann** [comment]
Hey, generally that makes sense.

However if I remember correctly you can not change it once it is set in the `setDragImage` function.
So to achieve this the best would probably be to let the user define a custom function for `_makeGhost` which returns an element.

You can not influence the opacity as far as I know. 

Does this approach make sense to you? Is this what you were suggesting?

I think it is more future prove to let the user define `makeGhost(draggedElement, sortable)` and do all the magic she wants return an element that will be set to the DragImage. The dragged and the sortable might be useful arguments, what do you think?

**Comment by lukasoppermann** [comment]
Hey, did you start on this already? I was thinking that maybe it would make sense to go all the way: hiding the ghost and add a custom elements that follows the mouse. This would enable to do more things on the future like animations. What do you think?

**Comment by mfriesen** [comment]
No I haven't done anything yet. That sounds pretty cool. I think that is beyond my javascript skills. For me just being able to customize the dragimage would be enough.

**Comment by lukasoppermann** [comment]
Hey @mfriesen, no problem. Feel free to implement the custom dragimage and we can enhance it later on if we want. It is probably good to have both options in any case, because there might be some issues with the advance version for some people.

**Comment by lukasoppermann** [comment]
This will fix #271
